### ISSUE TITLE
Check datatype for `missing_ranges` in `write_sav()`

### ISSUE THREAD

**Issue by Codeberg-AsGithubAlternative-buhtz** [Roche/pyreadstat#180]
- Create a `pandas.DataFrame` with a string column.
- Set `int(-9)` as a possible missing value in the `missing_ranges` parameter of `write_sav()`
- The `sav` file is created without any errors

The **problem** is that SPSS itself gives you an `Error 1405 in column 8` error.
I am not sure but I assume this is because the data type of the missing values does not fit to the data type of the column.

My experiments underpinning this. When you set `str(9)` as a missing value SPSS can open the file.

Solution: Check this data types while `write_sav()` and raise an TypeError.

To reproduce the problem
```
Python 3.9.2 (default, Feb 28 2021, 17:03:44)
[GCC 10.2.1 20210110] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import pandas as pd
>>> import pyreadstat as ps
>>> df = pd.DataFrame({'FOO': ['bar']})
>>> df
   FOO
0  bar
>>> ps.write_sav(df, 'foo.sav', missing_ranges={'FOO': [-9]})
```

Btw: I don't know why SPSS specifies "column 8" in the error messages even if there is only one column in the data.

**Comment by ofajardo** [comment]
makes sense, I was thinking on that check at some point and then came out of my radar. I will check my code to see if I can implement that.

**Comment by ofajardo** [comment]
I also just tried installing vanilla python 3.9 on my windows machine, just using all the defaults and then installed pyreadstat with pip. After that, pyreadstat was working out of the box and all the dlls are in the package folder (see screenshot, my username is deleted). So I really wonder how even to reproduce this issue to investigate it further. 

![image](https://user-images.githubusercontent.com/17498727/164501393-fd8ff422-8a9b-4eda-84b0-abda3b822e34.png)

**Comment by ofajardo** [comment]
solved in new version 1.1.5
